### PR TITLE
0.3: Fix compilation with no_std

### PR DESCRIPTION
**Pull request by ngg**
await! won't work with no_std, but at least the crates compile.

### CODE REVIEW

File: futures-core/src/lib.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ macro_rules! pin_mut {
         // Shadow the original binding so that it can't be directly accessed
         // ever again.
         #[allow(unused_mut)]
-        let mut $x = unsafe { ::std::mem::PinMut::new_unchecked(&mut $x) };
+        let mut $x = unsafe { ::core::mem::PinMut::new_unchecked(&mut $x) };
```
Comment (MajorBreakfast) [inline futures-core/src/lib.rs]: Could you implement it via a reexport of libcore and see if the strategy @cramertj proposes [here](https://github.com/rust-lang-nursery/futures-rs/pull/1051#discussion_r198942744) works?

Comment (ngg) [reply futures-core/src/lib.rs]: Yes, it seems to work.

File: futures-core/src/task/mod.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 //! Task notification.
 
+#[cfg(feature = "std")]
```
Comment (MajorBreakfast) [inline futures-core/src/task/mod.rs]: Maybe move this into `if_std! { ... }`

File: futures-executor/src/lib.rs [hunk 3]
```diff
@@ -13,9 +13,13 @@
 #[macro_use]
 extern crate std;
 
+#[cfg(feature = "std")]
 #[macro_use]
 extern crate futures_core;
 
+#[cfg(not(feature = "std"))]
+extern crate futures_core;
```
Comment (MajorBreakfast) [inline futures-executor/src/lib.rs]: Same here

File: futures-util/src/lib.rs [hunk 4]
```diff
@@ -31,9 +31,13 @@ macro_rules! if_std {
     )*)
 }
 
+#[cfg(feature = "std")]
```
Comment (MajorBreakfast) [inline futures-util/src/lib.rs]: And here :)

**Comment by MajorBreakfast** [comment]
I made a few notes. Thanks for fixing this! I also consider good `no_std` support important.

BTW `await` currently only works with `std` because it *currently* relies on thread local storage to access the `Context`.

**Comment by MajorBreakfast** [comment]
Could you apply the reexport thing also to `join!` and `select!`? Then it's applied to everything 🙂

https://github.com/rust-lang-nursery/futures-rs/tree/0.3/futures-util/src/await

**Comment by MajorBreakfast** [comment]
I must have missed that somehow. Good work!

BTW I‘m not authorited to merge PRs. We‘ll need to wait for @cramertj to do it. Likely on Monday

**Comment by cramertj** [comment]
This looks great, thanks so much!

**Pull request merged**